app/controllers/unattended_controller.rb

```diff
@@ -18,13 +21,23 @@ class UnattendedController < ApplicationController
   # all of our requests should be returned in text/plain
   after_action :set_content_type
 
-  # this actions is called by each operatingsystem post/finish script - it notify us that the OS installation is done.
   def built
-    logger.info "#{controller_name}: #{@host.name} is Built!"
+    logger.info "#{controller_name}: #{@host.name} is built!"
+    # clear possible previous errors
+    @host.build_errors = nil
     update_ip if Setting[:update_ip_from_built_request]
     head(@host.built ? :created : :conflict)
   end
 
+  def failed
```

Comment:
Should we allow this in `@spoof`?

Reply:
Sorry don't understand what you mean by "allow"?

Reply:
I mean that a logged in user can access the templates via unattended controller (click preview in the templates tab on host#show) even when the host is not in build mode. Should we add a guard that a user cannot access #failed and it's only allowed for hosts in build mode?

Reply:
Good point, although this does not render anything, but we should prevent from overwriting content in these cases. Thanks.